### PR TITLE
feature/update main

### PR DESCRIPTION
**Pull request by LeaBlancher**
# Implement navigations between screens and Refactor directory architecture and filenames

### PR description
This PR uses the already implemented NavigationAction class to navigate between the already implemented screens from the Sign-In and the Profile creation (both for Tutor and Student) to the overview screen.
It also rename some directories name and filenames ot have a clean arhcitecture with explicit name, that are coherent between them.

### Summary
- Update MainActivity to navigate between already implemented screens (AUTH, HOME, CREATE_PROFILE, CREATE_TUTOR_PROFILE. CREATE_TUTOR_SCHEDULE).
- Add a googleUid field in the Profile dataclass to be able to link the google account of the users with its profile.
- Add navigation and repository update for the CreateTutorProfileScreen and CreateTutorScheduleScreen when the user click on the validation button.

#### Pipeline of the navigation
First we have the Sign-In screen (AUTH). After having log-in with google sign-in, there is two paths: either we already have an profile and we directly go to the overview screen (HOME), or we need to create an account and we go to the create profile screen (CREATE_PROFILE).
In the create profile screen, we will need to enter the field to give our personnal information and then when we click on the validation button, we also have two paths. If we choose to be a student, we go to the overview screen (HOME). If we choose to be a tutor, we go to the create tutor profile screen (CREATE_TUTOR_PROFILE). After validation in this screen, we go to the create schedule for tutor screen (CREATE_TUTOR_SCHEDULE). After validating in this screen we finally go to the overview screen (HOME). 

### How to test
To test this PR, one can run the MainActivity and check that the pipeline describe before works and that the screen corresponds to the screenshots given below.
It could be useful two create to profiles, one which is a Tutor and one which is a Student.

### Screenshots
(In the order given: AUTH Screen (for Sign-In), HOME Screen (for now Greeting screen but will be update later), CREATE_PROFILE Screen, CREATE_TUTOR_PROFILE Screen, CREATE_TUTOR_SCHEDULE Screen
image image image image image

### CODE REVIEW

**Comment by alexs-rufer** [comment]
Great job on the implementation, thank you! I will test it in more details tomorrow, but I just noticed something that might be an issue. After creating a profile, it seems the binding with the Google ID isn't working. Although the account is successfully created in the database, I have to recreate it after logging in again.

File: app/src/androidTest/java/com/github/se/project/ui/authentification/SignInTest.kt [hunk 1]
```diff
@@ -1,10 +1,11 @@
-package com.github.se.project
+package com.github.se.project.ui.authentification
```
Comment (A-M-00) [inline app/src/androidTest/java/com/github/se/project/ui/authentification/SignInTest.kt:1]: Making test directories is a great initiative!

File: app/src/main/java/com/github/se/project/MainActivity.kt [hunk 2]
```diff
@@ -41,11 +43,17 @@ fun PocketTutorApp() {
...
 
+  // View models
   val authenticationViewModel: AuthenticationViewModel = viewModel()
 
   val listProfilesViewModel: ListProfilesViewModel =
       viewModel(factory = ListProfilesViewModel.Factory)
+  val profiles = listProfilesViewModel.profiles
 
+  // Google user unique id (as var to be able to pass from the SignIn to CreateProfile screens)
+  var googleUid = ""
```
Comment (A-M-00) [inline app/src/main/java/com/github/se/project/MainActivity.kt:54]: Saving the Uid globally here is smart, should minimise lookups

File: app/src/main/java/com/github/se/project/MainActivity.kt [hunk 3]
```diff
@@ -56,36 +64,44 @@ fun PocketTutorApp() {
...
-                    navigationActions.navigateTo(Screen.CREATE_PROFILE)
+                  onSuccess = { uid ->
+                    googleUid = uid
+                    val profile = profiles.value.find { it.googleUid == googleUid }
+
+                    if (profile != null) {
+                      // If the user already has a profile, navigate to the home screen
+                      listProfilesViewModel.setCurrentProfile(profile)
+                      navigationActions.navigateTo(Screen.HOME)
+                    } else {
```
Comment (A-M-00) [inline app/src/main/java/com/github/se/project/MainActivity.kt:75]: Very nice! Should allow for more thorough manual testing

File: app/src/main/java/com/github/se/project/ui/authentification/SignIn.kt [hunk 4]
```diff
@@ -1,4 +1,4 @@
-package com.github.se.project.screens
+package com.github.se.project.ui.authentification
```
Comment (A-M-00) [inline app/src/main/java/com/github/se/project/ui/authentification/SignIn.kt:1]: Certainly a better placement for this file!

File: app/src/main/java/com/github/se/project/ui/profile/CreateTutorProfile.kt [hunk 5]
```diff
@@ -23,13 +24,24 @@ import androidx.compose.ui.Modifier
...
 @Composable
-fun TutorInfoScreen(profile: Profile) {
+fun TutorInfoScreen(
+    navigationActions: NavigationActions,
+    listProfilesViewModel: ListProfilesViewModel =
+        viewModel(factory = ListProfilesViewModel.Factory)
+) {
+  val profile =
+      listProfilesViewModel.currentProfile.collectAsState().value
+          ?: return Text(text = "No Profile selected. Should not happen.", color = Color.Red)
```
Comment (A-M-00) [inline app/src/main/java/com/github/se/project/ui/profile/CreateTutorProfile.kt:43]: Putting safety tests is always a good initiative

File: app/src/main/java/com/github/se/project/ui/profile/CreateTutorSchedule.kt [hunk 6]
```diff
@@ -1,32 +1,61 @@
...
 @Composable
-fun AvailabilityScreen(profile: Profile, onProfileUpdate: (Profile) -> Unit) {
+fun AvailabilityScreen(
+    navigationActions: NavigationActions,
+    listProfilesViewModel: ListProfilesViewModel =
+        viewModel(factory = ListProfilesViewModel.Factory)
+) {
+  val profile =
+      listProfilesViewModel.currentProfile.collectAsState().value
+          ?: return Text(text = "No Profile selected. Should not happen.", color = Color.Red)
```
Comment (A-M-00) [inline app/src/main/java/com/github/se/project/ui/profile/CreateTutorSchedule.kt:55]: Same as above, good initiative!

File: app/src/main/java/com/github/se/project/ui/profile/CreateTutorProfile.kt
Comment (A-M-00) [inline app/src/main/java/com/github/se/project/ui/profile/CreateTutorProfile.kt:1]: Certainly a more intuitive name than the old one

**Review by A-M-00 — APPROVED** [review]
Great work overall!

**Pull request merged**